apps/src/sites/code.org/pages/views/workshop_search.js

```diff
@@ -10,30 +11,265 @@ const markerCustererOptions = {
   gridSize: 10
 };
 
-var gmap,
+var map,
   markersByLocation = {},
   infoWindow,
   markerClusterer;
 
 $(document).ready(function() {
-  initializeMap();
-  loadWorkshops();
+  if (window.location.search.includes('mapbox')) {
+    $('#gmap').hide();
+    $('#mapbox-map').show();
+    $('#geocomplete').hide();
+    $('#geocomplete-form').hide();
+    $('#mapboxgeocoder').show();
+    initializeMapboxMap();
+    map.on('load', loadMapboxWorkshops);
+  } else {
+    initializeGoogleMap();
+    loadGoogleMapWorkshops();
+  }
 });
 
-function initializeMap() {
+function initializeMapboxMap() {
+  map = new mapboxgl.Map({
+    container: 'mapbox-map',
+    style: 'mapbox://styles/mapbox/streets-v11',
+    center: [-95.665, 37.6],
+    zoom: 3
+  });
+
+  map.dragRotate.disable();
+  map.scrollZoom.disable();
+  map.dragPan.disable();
+  map.addControl(
+    new mapboxgl.NavigationControl({showCompass: false}),
+    'bottom-right'
+  );
+  const enableMouseControls = () => {
+    map.scrollZoom.enable();
+    map.dragPan.enable();
+  };
+  // Enable mouse controls when the map is clicked
+  map.on('click', enableMouseControls);
+  // Enable mouse controls when the zoom (+/-) buttons are pressed
+  map.on('zoom', enableMouseControls);
```

Comment:
Oh, I see.

Reply:
Yeah, this was a thing on the hour of code map where a map that's just in the middle of a page with a bunch of other stuff will capture the scrolling. This forces the user to interact with the map to be able to scroll-zoom.